src/transformers/modeling_utils.py

```diff
@@ -996,6 +997,9 @@ def _generate_beam_search(
                 # Compute next scores
                 next_scores = torch.gather(_scores, -1, next_tokens)  # (batch_size, num_beams * 2)
 
+                # sort the sampled vector to make sure that the first num_beams samples are the best
```

Comment:
I added this small sort function for the following reason. We are sampling `2 * num_beams` samples per `batch_idx` and always take the first three of those samples. I think the first three samples that we take should then also correspond to the best three out of 6 samples.